stan/math/opencl/matrix_cl.hpp

```diff
@@ -133,106 +218,17 @@ class matrix_cl {
                      a.rows(), "destination.rows()", rows());
     check_size_match("assignment of (OpenCL) matrices", "source.cols()",
                      a.cols(), "destination.cols()", cols());
+    // Need to wait for all of matrices events before destroying old buffer
```

Comment:
Is this comment out of date? I don't see code to do this anymore.

Reply:
The code below needed to be cleaned. I set this up now so that the old buffer is replaced only after read and write events happen for `this`